### PR TITLE
Fix cp dp differences

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review! Two changes are introduced:

1) The CP environment file includes the same version of `pytorch` as the DP environment file.

2) `idrstream_cp.py` now just copies the metadata file into the images directory instead of first saving the metadata csv into a different folder and copying it from there. More explanation about this change is given in my second comment.

### CODE REVIEW

File: example_notebooks/example_files/CP_files/mitocheck_idr_cp.cppipe
Comment (jenna-tomkinson) [inline example_notebooks/example_files/CP_files/mitocheck_idr_cp.cppipe]: @roshankern 

Wasn't this already changed in #18?

File: example_notebooks/example_scripts/example_cp.py
Comment (jenna-tomkinson) [inline example_notebooks/example_scripts/example_cp.py:59]: I like this change!

File: idrstream/CP_idr.py
Comment (jenna-tomkinson) [inline idrstream/CP_idr.py:142]: Since I would not think that every metadata file would be a `.tsv` file, this would seem redundant to convert the file to csv if it already is a csv. Maybe making this an `if` statement to assess if file ends with ".tsv" to then convert it.

File: idrstream/CP_idr.py
Comment (jenna-tomkinson) [inline idrstream/CP_idr.py:141]: Why was this deleted? Isn't this code what copied the file into the images directory? I could be missing it somewhere else in this file but after going over this function for `init_cp_run`, I don't see where anything was actually copied over.

File: idrstream/CP_idr.py
Comment (jenna-tomkinson) [inline idrstream/CP_idr.py:135]: Looks like you deleted the summary for the function and never added it back in. Consider adding this back in before merging so that this matches with the other functions.

File: idrstream/CP_idr.py
Comment (jenna-tomkinson) [inline idrstream/CP_idr.py]: Missing new line at the end of this file. Please add to remove the error.

File: idrstream_cp_env.yml
Comment (jenna-tomkinson) [inline idrstream_cp_env.yml:12]: Perfect! 

It looks like when I first made this implementation, I had added to the main README to install PyTorch after making the environment which I thought at the time was a good idea based on how we did it in [cell-health-data](https://github.com/WayScience/cell-health-data/tree/master/1.segment-data) where PyTorch was downloaded separately. Now with this issue with PyTorch causing segmentation issues, this is the more optimal method!

File: example_notebooks/example_scripts/example_cp.py
Comment (jenna-tomkinson) [inline example_notebooks/example_scripts/example_cp.py:91]: Since this has been changed to `init_cp_run`, I don't see it actually used in this py file or in the notebook. 

Without this function being initialized, then CellProfiler will not be able to compile any metadata because the file won't be located in the image directory.

File: example_notebooks/example_scripts/example_cp.py
Comment (jenna-tomkinson) [inline example_notebooks/example_scripts/example_cp.py]: For this .py file, I see that `In[3]` that the function `convert_tsv_to_csv` is being used. I am guessing this should be removed or changed.

File: idrstream_cp_env.yml
Comment (jenna-tomkinson) [reply idrstream_cp_env.yml:12]: Can you please delete the `PyTorch` portion of the README since an individual won't need to install PyTorch from their website anymore.

**Review by jenna-tomkinson — CHANGES_REQUESTED** [review]
This PR has many positive changes, especially in the environment for CellProfiler (Glad that you were able to find this!)

I do have comments about the changes you have made that I would like for you to address. These are mainly about the idrstream CellProfiler files.

One major point: In the CellProfiler example notebook, I do not see `init_cp_run` used so I worry that metadata compiling issues are occurring because of this.

File: example_notebooks/example_files/CP_files/mitocheck_idr_cp.cppipe
Comment (roshankern) [reply example_notebooks/example_files/CP_files/mitocheck_idr_cp.cppipe]: No, it looks like we both missed the second line where a regular expression with frames is used and only changed the first line. Below is a screenshot of the current version of [mitocheck_idr_cp.cppipe](https://github.com/WayScience/IDR_stream/blob/main/example_notebooks/example_files/CP_files/mitocheck_idr_cp.cppipe):
image

(Notice the very bottom line is still using the wrong regular expression for `Frames`)

Also, I missed the change to `?P<Plate>LT[0-9]{4}_[0-9]{2}` in this bottom line, which has been added in https://github.com/WayScience/IDR_stream/pull/21/commits/95cac866f2c8157207a41efac81d1aad1f8a1363

Comment (jenna-tomkinson) [reply example_notebooks/example_files/CP_files/mitocheck_idr_cp.cppipe]: Ahh good find! This is interesting that it shows this line twice and differently. This is an example of why I only change things within the GUI since I know for sure I changed it in the right parameter. Not sure why there are two lines, but this should still work!

File: idrstream_cp_env.yml
Comment (roshankern) [reply idrstream_cp_env.yml:12]: Done in https://github.com/WayScience/IDR_stream/pull/21/commits/ee3fb6a9eee0148fbffb05bfffe08056fae71e68!

File: idrstream/CP_idr.py
Comment (roshankern) [reply idrstream/CP_idr.py]: Done in https://github.com/WayScience/IDR_stream/pull/21/commits/ee3fb6a9eee0148fbffb05bfffe08056fae71e68!

File: idrstream/CP_idr.py
Comment (roshankern) [reply idrstream/CP_idr.py:135]: Done in https://github.com/WayScience/IDR_stream/pull/21/commits/ee3fb6a9eee0148fbffb05bfffe08056fae71e68!

File: idrstream/CP_idr.py
Comment (roshankern) [reply idrstream/CP_idr.py:142]: The comment for line 134 here is wrong. It should say `save metadata as csv so CellProfiler can read this metadata` (updated to say this in the most recent commit). `data_to_process` is a pandas dataframe and thus doesn't need to be checked to be a `.tsv`.

Comment (jenna-tomkinson) [reply idrstream/CP_idr.py:142]: Ahh okay that is much clearer!

**Comment by roshankern** [comment]
This comment includes extended documentation for the second change in this PR (`idrstream_cp.py` now just copies the metadata file into the images directory instead of first saving the metadata csv into a different folder and copying it from there):

CellProfiler (CP) requires a metadata `.csv` file in the `images` directory (within the CP project directory, within tmp directory of IDR_stream run). To follow `idrstream_dp.py` convention, we want the user to load in a metadata file (with locations, perturbation data, etc) as a pandas dataframe and pass this metadata dataframe into the `run_cp_stream()` method so IDR_stream has an idea of where the cells we desire features from are located.

 The current version of `idrstream_cp.py` requires a user to (see [example_cp.ipynb](https://github.com/WayScience/IDR_stream/blob/main/example_notebooks/example_cp.ipynb)):
 1) Use the `stream.convert_tsv_to_csv()` method to convert the `.tsv` metadata file into a `.csv` file, which will be saved in a directory not accessed by CellProfiler.
2) Use `stream.copy_CP_files()` to copy the saved `.csv` from step 1 into the `images` directory accessed by CellProfiler.
3) Load in the data to process (metadata) `.tsv` file as a pandas dataframe and pass this dataframe into `stream.run_cp_stream()` (by `idrstream_dp.py` convention).

The newer version of `idrstream_cp.py` requires a user to (see new [example_cp.ipynb](https://github.com/roshankern/IDR_stream/blob/fix-cp-dp-differences/example_notebooks/example_cp.ipynb)):
1) Simply load in the data to process (metadata) `.tsv` file as a pandas dataframe and pass this dataframe into `stream.run_cp_stream()` (by `idrstream_dp.py` convention).

Now, in the background, `idrstream_cp.py` will save the metadata pandas dataframe as a `.csv` in the `images` directory by itself (a helper function `init_cp_run(metadata_dataframe)` is called at the beginning of `stream.run_cp_stream()` to accomplish this task).

File: example_notebooks/example_scripts/example_cp.py
Comment (roshankern) [reply example_notebooks/example_scripts/example_cp.py:91]: `idrstream_cp` calls `init_cp_run` by itself to save the user the hassle! See my extended comment below for a more detailed explanation.

File: idrstream/CP_idr.py
Comment (roshankern) [reply idrstream/CP_idr.py:141]: No file needs to be copied. Instead, the dataframe is just saved as a `.csv` with `to_csv()`. Please see my comment below for a more detailed explanation.

File: example_notebooks/example_scripts/example_cp.py
Comment (roshankern) [reply example_notebooks/example_scripts/example_cp.py]: Yep it should be removed! Done in https://github.com/WayScience/IDR_stream/pull/21/commits/d4f0f47a9b7f6e0f0b663e79b609f2a361ba5d26

**Review by jenna-tomkinson — APPROVED** [review]
Thank you for addressing those comments, I think these change makes much more sense and do take a lot off of the user! My last recommendation would be that you change this documentation within the `init_cp_run` to be a bit clearer which I provide my change below:

```python
# make an images directory for the input of CellProfiler
metadata_save_path = pathlib.Path(
 f"{self.CP_project_path}/inputs/images/data_to_process.csv"
)
metadata_save_path.parents[0].mkdir(parents=True, exist_ok=True)
        
# save metadata to the images directory as csv for CellProfiler to read
data_to_process.to_csv(metadata_save_path, index=False)
```
Feel free to change this but the current iteration stating "make directory for the input of cellprofiler and copy metadata file into images folder" made it seem like there should be a copying function (e.g. shutil.copyfile) where that line is located.

LGTM after that!

**Pull request merged**
